File: TOI-Zero_68/A2/023/023.cpp
#include<bits/stdc++.h>
using namespace std;

#define ll long long

int main(){
	ios::sync_with_stdio(0);
	cin.tie(0);
	string s;
    cin >> s;
    for(int i=0;i<s.size();i++){
    	s[i]=toupper(s[i]);
	}
    int mx=0,cnt=0,sr=0,sb=0,ok=1,ccnt=0,id=0,unn=1;
    for(int i=0;i<s.size();i++){
        if(s[i]!='I' && s[i]!='T'){
        	unn=0;
		}
        if(sb && s[i]!='I' && s[i]!='T' && ccnt==0){
			ok=0; 
			break;
		}
        if(sr && s[i]!='A' && cnt==0){
			ok=0; 
			break;
		}
        if(s[i]=='R'){
        	sr=1;
        	cnt=0;
		}
        else if(s[i]=='A'){
            if(!sr){
				ok=0; 
				break;
			}
            cnt++;
            mx=max(mx,cnt);
        }
		else if(s[i]=='B'){
			sb=1;
			ccnt=0;
		}
        else if((s[i]=='I' || s[i]=='T') && sb){
        	ccnt++;
		}
        ++id;
    }
    if(!ok || (sr && cnt==0) || (sb && ccnt==0)){
    	cout << "no " << min(id,(int)s.size()-1);	
	}
    else if(unn){
    	cout << "unknown " << s.size();
	}
    else if(ok){
    	cout << "yes " << mx;
	}
}

File: TOI-Zero_68/A2/023/023.py
s = input().upper()

mx = 0
cnt = 0
sr = 0
sb = 0 
ok = 1 
ccnt = 0  
id = 0
unn = 1  

for i in range(len(s)):
    if s[i] not in ['I', 'T']:
        unn = 0

    if sb and s[i] not in ['I', 'T'] and ccnt == 0:
        ok = 0
        break

    if sr and s[i] != 'A' and cnt == 0:
        ok = 0
        break

    if s[i] == 'R':
        sr = 1
        cnt = 0

    elif s[i] == 'A':
        if not sr:
            ok = 0
            break
        cnt += 1
        mx = max(mx, cnt)

    elif s[i] == 'B':
        sb = 1
        ccnt = 0

    elif s[i] in ['I', 'T'] and sb:
        ccnt += 1

    id += 1

if not ok or (sr and cnt == 0) or (sb and ccnt == 0):
    print("no", min(id, len(s) - 1))

elif unn:
    print("unknown", len(s))

else:
    print("yes", mx)
# WRONG OUTPUT in Case 2 7 9
# RECOMMENDED C++
